internal/common/enums.rs

```diff
@@ -131,6 +131,20 @@ macro_rules! for_each_enums {
                 Action,
             }
 
+            /// This enum describes the different reasons for a FocusEvent
+            enum FocusEventReason {
+                /// The keyboard caused the event (tabbing)
+                Keyboard,
```

Comment:
Maybe `KeyboardNavigation` or `KeyboardShortcut` 
In slint it would then be `FocusReason.keyboard` or `FocusReason.keyboard-navigation` or `FocusReason.keyboard-shortcut`

Or maybe `TabKey` ?